source/KeyHandlers.js

```diff
@@ -306,9 +306,49 @@ var keyHandlers = {
                 self._updatePath( range, true );
             }
         }
-        // Otherwise, leave to browser but check afterwards whether it has
-        // left behind an empty inline tag.
+        // Nate: previously this was left to the browser but had issues with non-editable spans.  Furthermore
+        // firefox had an odd bug where it is confused by non-editable spans causing spaces to be added
+        // inside the non-editable block rather than deleting the appropriate character inside editable
+        // text.  There is some information on what is probably the same bug here:
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=685445
         else {
```

Comment:
at this else statement, most of the hard backspace stuff has been taken care of.  What is left is the simple case, delete the character directly behind the cursor, and the harder case of deleting noneditable elements altogether.  It took some time to get right.  There are tricky situations, it seems like you would just always take the text you are in and move back one then delete.  However, in something like: <b>bold</b>something, if your cursor is at the s, the naive previous element will be the <b> tag.  So you need to find the closest leaf of the tree, not only the previous node